validator/client/src/main/java/tech/pegasys/teku/validator/client/BeaconProposerPreparer.java

```diff
@@ -141,6 +147,28 @@ public Optional<Eth1Address> getFeeRecipient(final BLSPublicKey publicKey) {
         .or(() -> defaultFeeRecipient);
   }
 
+  @Override
+  public Optional<UInt64> getGasLimit(BLSPublicKey publicKey) {
+    if (validatorIndexCannotBeResolved(publicKey)) {
+      return Optional.empty();
+    }
+    return maybeProposerConfig
+        .flatMap(config -> getGasLimitFromProposerConfig(config, publicKey))
+        .or(() -> runtimeProposerConfig.getGasLimitForPubKey(publicKey))
+        .or(
+            () ->
+                maybeProposerConfig.flatMap(
+                    proposerConfigProvider ->
+                        proposerConfigProvider.getDefaultConfig().getBuilder().isPresent()
+                            ? proposerConfigProvider
+                                .getDefaultConfig()
```

Comment:
maybe make a function like getFeeRecipient on getDefaultConfig so you dont have to create a builder etc...